apm-agent-plugins/apm-logging-plugin/apm-logging-plugin-common/src/main/java/co/elastic/apm/agent/loginstr/error/LoggerErrorHelper.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.loginstr.error;
+
+import co.elastic.apm.agent.impl.Tracer;
+import co.elastic.apm.agent.impl.error.ErrorCapture;
+import co.elastic.apm.agent.sdk.state.CallDepth;
+
+import javax.annotation.Nullable;
+
+public class LoggerErrorHelper {
+
+    private final CallDepth callDepth;
+    private final Tracer tracer;
+
+    public LoggerErrorHelper(Class<?> adviceClass, Tracer tracer){
+        this.callDepth = CallDepth.get(adviceClass);
+        this.tracer = tracer;
+    }
+
+    @Nullable
+    public Object enter(@Nullable Throwable exception, Class<?> originClass){
+        if (!callDepth.isNestedCallAndIncrement()) {
+            if(exception != null){
+                ErrorCapture error = tracer.captureException(exception, tracer.getActive(), originClass.getClassLoader());
+                if (error != null) {
+                    error.activate();
+                }
+                return error;
+            }
+        }
+        return null;
+    }
+
+    public void exit(@Nullable Object errorCapture){
```

Comment:
Since it is outside the advice now, please add javadoc that this needs to be called with the value returned from the corresponding `enter` invocation.